src/use-queue.ts

```diff
@@ -0,0 +1,50 @@
+import { useState, useMemo } from 'react'
+import produce from 'immer'
+
+export const useQueue = <T>(initialValue: T[] = []) => {
+  const [value, setValue] = useState(initialValue)
+
+  const handlers = useMemo(
+    () => ({
+      dequeue: () => {
+        if (value.length === 0) return
+
+        let shifted
+        setValue((oldQueue) =>
+          produce(oldQueue, (draft: T[]) => {
+            shifted = draft.shift()
```

Comment:
... And used `.pop()` here so we can remove the last item. It's just personal preference, to be honest.
We can let it the way you made, but I've seen more examples for `unshift/pop` and I think people are more familiar with. Thoughts?

Reply:
I can change, no problems to me!